test_app/CMakeLists.txt

```diff
@@ -2,7 +2,16 @@
 # CMakeLists in this exact order for cmake to work correctly
 cmake_minimum_required(VERSION 3.5)
 
-set(EXTRA_COMPONENT_DIRS ../libsodium ../expat ../esp_encrypted_img)
+set(EXTRA_COMPONENT_DIRS ../libsodium ../expat ../esp_encrypted_img ../cbor ../jsmn ../qrcode)
+
+if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_GREATER_EQUAL "4.4")
+    list(APPEND EXTRA_COMPONENT_DIRS ../pid_ctrl)
+endif()
+
+if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_GREATER_EQUAL "5.0")
+    list(APPEND EXTRA_COMPONENT_DIRS ../sh2lib ../nghttp)
+endif()
+
 
 # Set the components to include the tests for.
 set(TEST_COMPONENTS libsodium expat esp_encrypted_img CACHE STRING "List of components to test")
```

Comment:
I wonder what would happen if we set TEST_COMPONENTS equal to EXTRA_COMPONENT_DIRS (stripping the `../` prefix from each).
Would the build system complain that some components don't actually have a test subdirectory?

If it wouldn't this would be a convenient way to not forget to add new components to TEST_COMPONENTS later.

Reply:
This is a very good idea. The build system doesn't mind the non-existent `test` folder.

CMake updated.